sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -123,4 +138,30 @@ public NettyAsyncHttpClientBuilder setNioEventLoopGroup(NioEventLoopGroup nioEve
         this.nioEventLoopGroup = nioEventLoopGroup;
         return this;
     }
+
+    /**
+     * Sets the connection pool strategy the client should use, which by default will be set to a fixed pool with
+     * maximum number of connections equal to {@code max(logical processor count, 8) * 2}.
+     * @param connectionProvider the connection provider defining the connection pool strategy
+     * @return the updated NettyAsyncHttpClientBuilder object
+     */
+    public NettyAsyncHttpClientBuilder connectionProvider(ConnectionProvider connectionProvider) {
+        this.connectionProvider = connectionProvider;
+        return this;
+    }
+
+    /**
+     * Sets custom configurations for the reactor netty client directly on the reactor netty {@link HttpClient}. The
+     * provided configurations in this method is applied first, before the other convenience methods, thus having the
+     * least priority and may be overwritten. For example, you can set raw configurations as shown below:
+     *
+     * {@codesnippet com.azure.core.http.netty.NettyAsyncHttpClientBuilder#reactorNettyConfiguration}
+     *
+     * @param rawConfiguration configurations on the raw reactor netty {@link HttpClient}
+     * @return the updated NettyAsyncHttpClientBuilder object
+     */
```

Comment:
I'm slightly worried about the low precedence of this compared with the overwriting that will occur for port, wiretap, and tcpConfiguration in the build method. I would put this at a higher precedence - what do you think?

Reply:
I guess you are probably right. Users will probably use this as their last advanced resort and the configurations made here are probably later than the other convenience methods.